relay/pkg/btc/remote/remote.go

```diff
@@ -140,6 +146,41 @@ func serializeHeader(header *wire.BlockHeader) ([]byte, error) {
 func (rc *remoteChain) GetHeaderByDigest(
 	digest btc.Digest,
 ) (*btc.Header, error) {
-	// TODO: implementation
-	return nil, nil
+
+	blockHeader, err := rc.client.GetBlockHeader((*chainhash.Hash)(&digest))
+	if err != nil {
+		return nil, fmt.Errorf(
+			"could not get block header for hash [%s]: [%v]",
+			digest.String(),
+			err,
+		)
+	}
+
+	headerVerbose, err := rc.client.GetBlockHeaderVerbose((*chainhash.Hash)(&digest))
```

Comment:
I didn't find any cleaner btc api method for returning block height based on block hash.

I see one alternative:
Instead of calling `GetBlockHeader` and `GetBlockHeaderVerbose` we could just call `GetBlockHeaderVerbose` as it contains all the information we need.
But `GetBlockHeaderVerbose` returns data in not very useful format (for example hashes are returned as strings).

We would have to construct `wire.BlockHeader` structure (which is needed for serialization) from the object returned from `GetBlockHeaderVerbose`  which would probably add more code.

Reply:
Ok fine. I think we can leave it as is.